catchup/universalFetcher.go

```diff
@@ -270,3 +270,99 @@ func (hf *HTTPFetcher) getBlockBytes(ctx context.Context, r basics.Round) (data
 func (hf *HTTPFetcher) address() string {
 	return hf.rootURL
 }
+
+type wrongCertFromPeerError struct {
```

Comment:
note that the common naming conventions for errors are "ErrXXX". in your case, leave it with lower-case, but still name it `errWrongCertFromPeer` rather then `wrongCertFromPeerError`